components/eventForm/EventForm.tsx

```diff
@@ -0,0 +1,178 @@
+import { useReactHookFormContext } from "../../hooks/FormHooks"
+import { EditEventInput } from "../../lib/events"
+import React, {
+  ReactNode,
+  useContext,
+  createContext,
+  useState,
+  useRef,
+  useEffect
+} from "react"
+import { useForm, FormProvider, useController } from "react-hook-form"
+import { Keyboard } from "react-native"
+import { EventFormValues } from "./EventFormValues"
+
+export type EventFormSection = "date" | "color" | "advanced"
+
+/**
+ * Props for `EventForm`.
+ */
+export type EventFormProps = {
+  /**
+   * The initial values to use in the form, updates to this field
+   * will not update the current form values.
```

Comment:
Could we make it readonly?

Reply:
There wouldn't really be much of a reason to do so, since you can't really modify props from inside a component...